src/js/view/controls/components/menu/menu.js

```diff
@@ -0,0 +1,499 @@
+                    argument = item;
+                    content = isRtl(options.tooltipText) ? 'x' + item : item + 'x';
+                    break;
+                case 'audioTracks':
+                    content = item.name;
+                    break;
+                default:
+                    break;
+            }
+            const menuItem = new Item(content, (evt) => {
+                if (this.active) {
+                    return;
+                }
+                action(argument || index);
+                this.items.filter(sibling => sibling.active === true).forEach(activeItem => {
+                    activeItem.deactivate();
+                });
+                menuItem.activate();
+                this.mainMenu.close(evt);
+            });
+
+            menuItem.ui.on('keydown', (event) => {
+                const focusElement = (ele, i) => {
+                    if (ele) {
+                        ele.focus();
+                    } else if (i !== undefined) {
+                        menuItems[i].el.focus();
+                    }
+                };
+                const evt = event.sourceEvent;
+                const nextItem = nextSibling(categoryButtonElement);
+                const prevItem = previousSibling(categoryButtonElement);
+                const nextSubItem = nextSibling(evt.target);
+                const prevSubItem = previousSibling(evt.target);
+                const key = evt.key.replace(/(Arrow|ape)/, '');
+        
+                switch (key) {
+                    case 'Tab':
+                        focusElement(evt.shiftKey ? prevItem : nextItem);
+                        break;
+                    case 'Left':
+                        focusElement(prevItem || 
+                            previousSibling(document.getElementsByClassName('jw-icon-settings')[0]));
+                        break;
+                    case 'Up':
+                        focusElement(prevSubItem, menuItems.length - 1);
+                        break;
+                    case 'Right':
+                        focusElement(nextItem);
+                        break;
+                    case 'Down':
+                        focusElement(nextSubItem, 0);
+                        break;
+                    default:
+                        break;
+                }
+                evt.preventDefault();
+                if (key !== 'Esc') {
+                    // only bubble event if esc key was pressed
+                    evt.stopPropagation();
+                }
+                
+            });
+
+            return menuItem;
+        });
+
+        return menuItems;
+    }
+    defaultChild() {
+        return this.children.quality || this.children.captions || this.children.audioTracks || this.children.sharing;
+    }
+    setMenuItems(menuItems, initialSelectionIndex) {
+        if (!menuItems) {
+            this.removeMenu();
+        } else {
+            this.items = [];
```

Comment:
small thing that is probably me overoptimizing: any use of this and the below line residing in a `resetMenu` function?

Reply:
would make it more testable 😬 but if this is the only occurrence of this behavior, no sense in pulling it out